### DISCUSSION TITLE
Get Version for backup bash script

### DISCUSSION THREAD

**Discussion by robinsch1988** [ampache/ampache#3528] [Q&A] [answered]
Hello everyone,
I want to automate my ampache backup bash script and need to get the current version of ampache via a sql query or a file to put it in the backup filename.

I have already looked in the database in tables like "update_info", but cannot find a corresponding value.
 
Also looked into the code, but my php skills are very very rusty.
I got so far:
In the file "/path/to/ampache/public/templates/footer.inc.php" the ampache version is set with "AmpConfig::get('version')". The "AmpConfig" class (located in /path/to/ampache/src/Config/AmpConfig.php) use an global array called "$_global". But I cannot find the file where this key value pair like "$_global['version'] = '5.6.0' " is set.
Maybe you can push me in the right direction. Thank you in advance.

**Comment by lachlan-00** [comment]
i'll put a command into the cli command to just print the version.

that way you can just use the cli to print the version.

i guess in bash/dash it would be something like
`VERSION={php /var/www/ampache/bin/cli -v}`

I'll do it now cause that's actually something we should have and it will go in ampache 5 and 6

**Reply by lachlan-00** [reply]
coming soon to 5/6

![image](https://github.com/ampache/ampache/assets/1305249/36613b17-b3fb-4cfb-8696-088db54a878f)

**Comment by lachlan-00** [comment, marked as answer]
What i do in my current bash scripts is grep the file with the config version

```
RELEASEVERSION=`grep -oP '[0-9]+\.[0-9]+\.[0-9]+' ./ampache-patch6/src/Config/Init/InitializationHandlerConfig.php`
```

does the same thing as the cli version above
![image](https://github.com/ampache/ampache/assets/1305249/056f1dec-bf2e-4baa-bc6d-57e0c748a557)

here's the release file i use for scripting zip builds
https://github.com/lachlan-00/ampache-administrator/blob/master/build_release.sh

**Comment by robinsch1988** [comment]
Thank you. The grep command works perfect.

**Reply by robinsch1988** [reply]
The grep command was just a temporary solution until version 5.6.2.
Since 5.6.2 you can use the `cli` command to get the version.
Here is an example if you use Apache as a webserver with the user `www-data`:
```
john@hostname123:~$ sudo -u www-data php /var/www/html/ampache/bin/cli show:version
5.6.2-release
```
